apps/portal-front/src/components/unenroll/octi/index.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+  CanUnenrollOCTIInstanceFragment,
+  UnenrollOCTIInstance,
+} from '@/components/enroll/enroll.graphql';
+import { EnrollStateLayout } from '@/components/enroll/state/layout';
+import Loader from '@/components/loader';
+import { UnenrollOCTIConfirm } from '@/components/unenroll/octi/confirm';
+import { UnenrollOCTIInstanceNotEnrolled } from '@/components/unenroll/octi/instance-not-enrolled';
+import { UnenrollOCTIMissingCapability } from '@/components/unenroll/octi/missing-capability';
+import useMountingLoader from '@/hooks/useMountingLoader';
+import { enrollCanUnenrollOCTIInstanceFragment$key } from '@generated/enrollCanUnenrollOCTIInstanceFragment.graphql';
+import EnrollCanUnenrollOCTIInstanceQueryGraphql, {
+  enrollCanUnenrollOCTIInstanceQuery,
+} from '@generated/enrollCanUnenrollOCTIInstanceQuery.graphql';
+import { enrollUnenrollOCTIInstanceMutation } from '@generated/enrollUnenrollOCTIInstanceMutation.graphql';
+import UserListOrganizationAdministratorsQueryGraphql, {
+  userListOrganizationAdministratorsQuery,
+} from '@generated/userListOrganizationAdministratorsQuery.graphql';
+import { toast } from 'filigran-ui/clients';
+import { useTranslations } from 'next-intl';
+import React, { useState } from 'react';
+import {
+  PreloadedQuery,
+  useFragment,
+  useMutation,
+  usePreloadedQuery,
+  useQueryLoader,
+} from 'react-relay';
+
+interface Props {
+  platformId: string;
+  queryRef: PreloadedQuery<enrollCanUnenrollOCTIInstanceQuery>;
+}
+
+type UnenrollmentStatus = 'idle' | 'succeeded' | 'failed';
+
+export const UnenrollOCTI: React.FC<Props> = ({ queryRef, platformId }) => {
+  const t = useTranslations();
+  const canUnenrollPreloadedQuery =
+    usePreloadedQuery<enrollCanUnenrollOCTIInstanceQuery>(
+      EnrollCanUnenrollOCTIInstanceQueryGraphql,
+      queryRef
+    );
+
+  const { instance } = useFragment<enrollCanUnenrollOCTIInstanceFragment$key>(
+    CanUnenrollOCTIInstanceFragment,
+    canUnenrollPreloadedQuery.canUnenrollOCTIInstance
+  );
+
+  const [
+    organizationAdministratorsQueryRef,
+    loadOrganizationAdministratorsQuery,
+  ] = useQueryLoader<userListOrganizationAdministratorsQuery>(
+    UserListOrganizationAdministratorsQueryGraphql
+  );
+  useMountingLoader(loadOrganizationAdministratorsQuery, {
+    organizationId: instance?.organizationId,
+  });
+  const [unenrollInstance] =
+    useMutation<enrollUnenrollOCTIInstanceMutation>(UnenrollOCTIInstance);
+
+  const [status, setStatus] = useState<UnenrollmentStatus>('idle');
+  const cancel = () => {
+    window.opener?.postMessage({ action: 'cancel' }, '*');
+  };
+
+  const confirm = () => {
```

Comment:
nitpick: declare these functions after the set of "if" instructions to avoid memory allocation of useless methods

Reply:
I think that changing the order of `ifs` and function declaration will decrease readability of this component.
I also think that the memory allocation of two methods does not have a huge impact. 
We could only move two ifs above this methods because the other needs it :/ 

Do you agree ? 🙂